### PR TITLE
Add log-rotation-size option

### PR DESCRIPTION
**Pull request by yamatcha**
ref: #104

### CODE REVIEW

File: cmd/moco-agent/cmd/root.go [hunk 1]
```diff
@@ -142,6 +143,20 @@ var rootCmd = &cobra.Command{
 			}
 		}()
 
+		// Rotate if the file size exceeds logRotationSize
+		ticker := time.NewTicker(time.Second)
```
Comment (shunki-fujita) [inline cmd/moco-agent/cmd/root.go:147]: This ticker should be written in the range of if statements

Comment (shunki-fujita) [reply cmd/moco-agent/cmd/root.go:147]:
```suggestion
		if config.logRotationSize > 0 {
			ticker := time.NewTicker(time.Second)
			defer ticker.Stop()
			go func() {
				...
```

File: server/rotate.go [hunk 2]
```diff
@@ -25,20 +25,58 @@ func (a *Agent) RotateLog() {
...
 	}
 
 	durationSeconds := time.Since(startTime).Seconds()
 	metrics.LogRotationDurationSeconds.Observe(durationSeconds)
 }
+
+// RotateLogIfSizeExceeded rotates log files if it exceeds rotationSize
+func (a *Agent) RotateLogIfSizeExceeded(rotationSize int64) {
+	errFile := filepath.Join(a.logDir, mocoagent.MySQLErrorLogName)
+	errFileStat, err := os.Stat(errFile)
```
Comment (shunki-fujita) [inline server/rotate.go:66]: Also check for the existence of files.

File: server/rotate.go [hunk 3]
```diff
@@ -25,20 +25,58 @@ func (a *Agent) RotateLog() {
...
+	if err != nil {
+		a.logger.Error(err, "failed to get stat of error log file")
```
Comment (shunki-fujita) [inline server/rotate.go:68]: return?

File: server/rotate.go [hunk 4]
```diff
@@ -25,20 +25,58 @@ func (a *Agent) RotateLog() {
...
+	}
+	if errFileStat.Size() > rotationSize {
+		a.RotateErrorLog()
+	}
+
+	slowFile := filepath.Join(a.logDir, mocoagent.MySQLSlowLogName)
+	slowFileStat, err := os.Stat(slowFile)
```
Comment (shunki-fujita) [inline server/rotate.go:75]: ditto

File: cmd/moco-agent/cmd/root.go [hunk 5]
```diff
@@ -142,6 +143,20 @@ var rootCmd = &cobra.Command{
...
+		if config.logRotationSize > 0 {
+			go func() {
```
Comment (shunki-fujita) [inline cmd/moco-agent/cmd/root.go:149]: No need for loops?

File: server/rotate.go (hunk 2 above)
Comment (yamatcha) [reply server/rotate.go:66]: When the file doesn't exists, os.Stat() returns error.
So I think checking existence of the file is not necessary.

**Review by shunki-fujita — APPROVED** [review]

**Pull request merged**